src/Async/Qu.swift

```diff
@@ -0,0 +1,200 @@
+//
+//  Qu.swift
+//
+//  Created by Logan Wright on 10/24/15.
+//  Copyright © 2015 lowriDevs. All rights reserved.
+//
+
+public typealias Block = () -> ()
+
+// MARK: Qu
+
+public class Qu {
+    
+    // MARK: Priority Enumeration
+    
+    public enum Priority {
+        case Background
+        case Main
+        case Custom(NSOperationQueue)
+        
+        var queue: NSOperationQueue {
+            switch self {
+            case .Background:
+                return NSOperationQueue()
+            case .Main:
+                return NSOperationQueue.mainQueue()
+            case .Custom(let customQueue):
+                return customQueue
+            }
+        }
+    }
+    
+    public enum Dependency {
+        case Last
+        case All
+        case Previous(Int)
+    }
+    
+    // MARK: Properties
+    
+    private(set) var priority: Priority
+    private(set) var operationQueue: NSOperationQueue
+    
+    /// Completion operation that will attempt to wait for other operations to finish and then execute.  If all operations have already executed before this operation is added, it will run immediately.  Even if you add operations afterwards.
+    private(set) var completion: Operation? {
+        willSet {
+            if completion != nil && newValue != nil {
+                print("*** Warning *** Setting multiple completion blocks will result in unpredictable behavior!")
+            }
+        }
+    }
+    
+    // MARK: Initialization
+    
+    public required init(priority: Priority) {
+        self.priority = priority
+        self.operationQueue = priority.queue
+    }
+    
+    // MARK: Class Functions
+    
+    class func Background(block: Block) -> Self {
+        let q = self.init(priority: .Background)
+        return q.Run(block)
+    }
+    
+    class func Main(block: Block) -> Self {
+        let q = self.init(priority: .Main)
+        return q.Run(block)
+    }
+    
+    class func Custom(queue: NSOperationQueue, block: Block) -> Self {
+        let q = self.init(priority: .Custom(queue))
+        return q.Run(block)
+    }
+    
+    // MARK: Dispatching Functions
+    
+    func Run(block: Block) -> Self {
+        return queue(block)
+    }
+    
+    func Also(block: Block) -> Self {
+        return queue(block)
+    }
+    
+    func Then(block: Block) -> Self {
+        return ThenAfter(.Last, block: block)
+    }
+    
+    func ThenAfter(dependency: Dependency, block: Block) -> Self {
+        let blockOp = Operation(block: block)
+        switch dependency {
+        case .Last:
```

Comment:
This is how switch statements are auto-formatted in Swift, I generally don't fight it :)